packages/lesswrong/components/admin/CurationNoticesItem.tsx

```diff
@@ -0,0 +1,192 @@
+  root: {
+    border: theme.palette.border.commentBorder,
+    borderRadius: isFriendlyUI ? theme.borderRadius.small : undefined,
+    cursor: "default",
+    marginBottom: 20,
+    background: theme.palette.background.pageActiveAreaBackground,
+    padding: 12,
+    position: "relative",
+    paddingBottom: 24,
+  },
+  editButton: {
+    ...theme.typography.body2,
+    color: theme.palette.primary.main,
+    padding: "0px 16px",
+    fontSize: "16px",
+    position: "absolute",
+    right: 10,
+    top: 10,
+    textTransform: "uppercase",
+    "&:hover": {
+      opacity: 0.5,
+      },
+},
+  publishButton: {
+      ...theme.typography.body2,
+      color: theme.palette.primary.main,
+      padding: "0px 16px",
+      fontSize: "16px",
+      position: "absolute",
+      right: 10,
+      bottom: 10,
+      textTransform: "uppercase",
+      "&:hover": {
+        opacity: 0.5,
+      },
+  },
+  meta: {
+    "& > div": {
+      marginRight: 5,
+    },
+    position: "relative",
+    display: "flex",
+    flexWrap: "wrap",
+    alignItems: "center",
+    justifyContent: "space-between",
+    rowGap: "6px",
+    marginBottom: 8,
+    color: theme.palette.text.dim,
+    paddingTop: "0.6em",
+    marginRight: isFriendlyUI ? 40 : 20,
+
+    "& a:hover, & a:active": {
+      textDecoration: "none",
+      color: isFriendlyUI ? undefined : `${theme.palette.linkHover.dim} !important`,
+    },
+  },
+  postTitle: {
+    ...theme.typography.body2,
+    ...theme.typography.commentStyle,
+    color: theme.palette.link.dim2,
+  },
+  username: {
+    ...theme.typography.body2,
+    ...theme.typography.commentStyle,
+    fontWeight: 600,
+    marginLeft: 25,
+  },
+  rightGroup: {
+    display: "flex",
+    alignItems: "center",
+    gap: 10,
+  },
+  commentBody: {
+    ...commentBodyStyles(theme)
+  }
+});
+
+const { ContentItemBody, Button, BasicFormStyles, WrappedSmartForm } = Components;
+
+export const CurationNoticesItem = ({curationNotice, classes}: {
+  curationNotice: CurationNoticesFragment,
+  classes: ClassesType<typeof styles>
+}) => {
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+
+  const [ edit, setEdit ] = useState<boolean>(false)
+
+  const { create } = useCreate({
+    collectionName: "Comments",
+    fragmentName: 'CommentsList'
+  });
+
+  const {mutate: UpdateCurrentCurationNotice} = useUpdate({
+    collectionName: "CurationNotices",
+    fragmentName: 'CurationNoticesFragment',
+  });
+
+  const makeComment = async (curationNotice: CurationNoticesFragment
+  ) => {
+    if (!curationNotice.contents) {throw Error("Error creating comment")}
+
+    const comment = {
+      postId: curationNotice.postId,
+      userId: curationNotice.userId,
+      contents: { originalContents: { 
+        data: curationNotice.contents.originalContents.data,
+        type: curationNotice.contents.originalContents.type,
+      }} as EditableFieldContents
+    };
```

Comment:
The idiomatic way to do this would be something like:

```
const { contents, postId, userId } = curatoinNotice;

if (!contents) {
  throw new Error('Curation notice is missing contents');
}

const { originalContents: { data, type } } = contents;

const comment = {
  postId,
  userId,
  contents: {
    originalContents: { data, type }
  }
};
```

(Also, is `contents` ever going to be missing?)

Reply:
edited thx! it didn't like your data/type syntax, so i left it as 

const comment = {
      postId,
      userId,
      contents: { originalContents: { 
        data: contents.originalContents.data,
        type: contents.originalContents.type,
      }} as EditableFieldContents
    };

also if i remove the as EditableFieldContents then the later use of comment in the useCreate function throws a type error :P

Reply:
nvm, done with robert now (i missed a line of his code, also did need the typecast)